awscli/customizations/rekognition.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.arguments import CustomArgument
+
+
+IMAGE_FILE_DOCSTRING = ('<p>The path to the image file you are uploading. '
+                        'Example: fileb://image.png</p>')
+IMAGE_DOCSTRING_ADDENDUM = ('<p>To specify a local file use <code>--%s</code> '
+                            'instead.</p>')
+
+
+class ParameterReplacement(object):
+    def __init__(self, source_param, target_param):
+        self.source_param = source_param
+        self.target_param = target_param
+
+
+FILE_PARAMETER_UPDATES = {
+    'compare-faces': [
+        ParameterReplacement('source-image', 'SourceImage'),
+        ParameterReplacement('target-image', 'TargetImage')
+    ],
+    'detect-faces': [ParameterReplacement('image', 'Image')],
+    'detect-labels': [ParameterReplacement('image', 'Image')],
+    'detect-moderation-labels': [ParameterReplacement('image', 'Image')],
+    'detect-text': [ParameterReplacement('image', 'Image')],
+    'index-faces': [ParameterReplacement('image', 'Image')],
+    'recognize-celebrities': [ParameterReplacement('image', 'Image')],
+    'search-faces-by-image': [ParameterReplacement('image', 'Image')],
```

Comment:
That was my original plan. I went this route since it is simpler. I don't really have a strong opinion about this so I can be swayed to automatic vs partially automatic with the extra cease of `compare-faces` vs hard coded as it is now.